pandas/core/frame.py

```diff
@@ -2216,10 +2216,26 @@ def to_feather(self, path, **kwargs) -> None:
         """,
     )
     def to_markdown(
-        self, buf: Optional[IO[str]] = None, mode: Optional[str] = None, **kwargs
+        self,
+        buf: Optional[IO[str]] = None,
+        mode: Optional[str] = None,
+        index: Optional[bool] = True,
+        **kwargs,
     ) -> Optional[str]:
+        if "showindex" in kwargs:
+            index = kwargs["showindex"]
+            del kwargs["showindex"]
+
+            warnings.warn(
+                "'showindex' is deprecated. Only 'index' will be used "
+                "in a future version. Use 'index' to silence this warning.",
+                FutureWarning,
+                stacklevel=2,
+            )
+
         kwargs.setdefault("headers", "keys")
         kwargs.setdefault("tablefmt", "pipe")
+        kwargs.setdefault("showindex", True if index is None else index)
```

Comment:
index should never be None, parameter type is bool, with default True.

so this could be just `kwargs.setdefault("showindex", index)`?